src/shared/modules/auth/index.js

```diff
@@ -0,0 +1,242 @@
+      input.setAttribute('type', 'hidden')
+      input.setAttribute('name', param)
+      input.setAttribute('value', params[param])
+      form.appendChild(input)
+    }
+
+    document.body.appendChild(form)
+    form.submit()
+  }
+
+  if (selectedSSOProvider.auth_flow === PKCE) {
+    const codeChallengeMethod =
+      ssoConfig.code_challenge_method || defaultCodeChallengeMethod
+    authLog(
+      `Auth flow "PKCE", using code_challenge_method: "${codeChallengeMethod}"`
+    )
+
+    const codeVerifier = createCodeVerifier(codeChallengeMethod)
+    window.sessionStorage.setItem(AUTH_STORAGE_CODE_VERIFIER, codeVerifier)
+
+    createCodeChallenge(codeChallengeMethod, codeVerifier).then(
+      codeChallenge => {
+        params = {
+          ...params,
+          code_challenge_method: codeChallengeMethod,
+          code_challenge: codeChallenge
+        }
+        _submitForm(form, params)
+      }
+    )
+  } else if (selectedSSOProvider.auth_flow === IMPLICIT) {
+    authLog('Auth flow "implicit flow"')
+    _submitForm(form, params)
+  } else {
+    const error = `Auth flow "${selectedSSOProvider.auth_flow}" is not supported.`
+    authLog(error)
+    return error
+  }
+}
+
+export const handleAuthFromRedirect = () =>
+  new Promise((resolve, reject) => {
+    const {
+      idp_id: idpId,
+      token_type: tokenType,
+      access_token: accessToken,
+      id_token: idToken,
+      error_description: errorDescription,
+      code,
+      state,
+      error
+    } = getInitialisationParameters()
+
+    removeSearchParamsInBrowserHistory(searchParamsToRemoveAfterAuthRedirect)
+
+    if (error) {
+      const errorMsg = `Error detected after auth redirect, aborting. Error: ${error}, Error description: ${errorDescription}`
+      authLog(errorMsg, 'warn')
+      reject(errorMsg)
+      return
+    }
+
+    if (!idpId) {
+      const errorIdpMsg = 'Invalid idp_id parameter, aborting'
+      authLog(errorIdpMsg, 'warn')
+      reject(errorIdpMsg)
+      return
+    }
+
+    const savedState = window.sessionStorage.getItem(AUTH_STORAGE_STATE)
+    if (state !== savedState) {
+      const errorStateMsg = 'Invalid state parameter, aborting'
+      authLog(errorStateMsg, 'warn')
+      reject(errorStateMsg)
+      return
+    }
+    window.sessionStorage.setItem(AUTH_STORAGE_STATE, '')
+
+    if ((tokenType || '').toLowerCase() === BEARER && accessToken) {
+      authLog('Successfully aquired access_token in "implicit flow"')
+
+      authDebug('Implicit flow id_token', idToken)
+      authDebug('Implicit flow access_token', accessToken)
+
+      const credentials = getCredentialsFromAuthResult(
+        { access_token: accessToken, id_token: idToken },
+        idpId
+      )
+      resolve(credentials)
+    } else {
+      authLog('Attempting to fetch token information in "PKCE flow"')
+
+      authRequestForToken(idpId, code)
+        .then(data => {
+          data.json().then(result => {
```

Comment:
missing the catch here (for `data.json()`)

Reply:
Letting the outer catch handle this